### PR TITLE
chore: streamline error checking process

### PR DESCRIPTION
**Pull request by bisakhmondal**
Signed-off-by: Bisakh Mondal <bisakhmondal00@gmail.com>

### What this PR does / why we need it:

https://github.com/apache/apisix/pull/6037#discussion_r779988474
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/openid-connect.lua [hunk 1]
```diff
@@ -293,30 +293,28 @@ function _M.rewrite(plugin_conf, ctx)
         -- the configured redirect URI after successful authentication.
         response, err = openidc.authenticate(conf)
 
-        if err then
+        if not response then
```
Comment (spacewander) [inline apisix/plugins/openid-connect.lua:296]: the openidc returns non-nil err to indicate something wrong, see:
https://github.com/zmartzone/lua-resty-openidc#sample-configuration-for-google-signin
https://github.com/zmartzone/lua-resty-openidc/blob/master/lib/resty/openidc.lua#L1494

File: t/plugin/basic-auth.t [hunk 2]
```diff
@@ -398,7 +398,7 @@ GET /t
 
 
 
-=== TEST 15: enable basic auth plugin using admin api, set hide_credentials = true
+=== TEST 18: enable basic auth plugin using admin api, set hide_credentials = true
```
Comment (tzssangglass) [inline t/plugin/basic-auth.t:401]: Need to fetch the latest code for the master branch, which has been fixed.

File: apisix/plugins/openid-connect.lua (hunk 1 above)
Comment (bisakhmondal) [reply apisix/plugins/openid-connect.lua]: Thanks. Updated

File: t/plugin/basic-auth.t (hunk 2 above)
Comment (bisakhmondal) [reply t/plugin/basic-auth.t]: Done. Thanks

**Review by tzssangglass — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**